cluster-healthcheck.metrics.md

```diff
@@ -0,0 +1,108 @@
+* **Notification**:
+  * Send a warning when `ActiveControllerCount != 1`
+  * Send an alarm when `ActiveControllerCount != 1` for more than 10s.
+
+## Offline partitions
+* **Type**: Message Delivery
+* **Description**:  An _Offline Partition_ is a partition without active leader and are hence not writable or readable. The presence of Offline partitions compromise the data integrity of the cluster.
+* **Metric**:  JMX `kafka.controller:type=KafkaController,name=OfflinePartitionsCount`
+* **Notification**: Alarm when `OfflinePartitionsCount > 0`
+
+## Unclean Elections
+* **Type**: Message Delivery
+* **Description**: Normally, when a broker that is the leader for a partition goes offline, a new leader is elected from the set of ISRs for the partition. An unclean leader election is a special case in which no available replicas are in sync. Because each topic _must_ have a leader, an election is held among the out-of-sync replicas and a leader is chosen—meaning any messages that were not synced prior to the loss of the former leader are lost forever. Essentially, unclean leader elections sacrifice consistency for availability.
+* **Metric**:  JMX `kafka.controller:type=ControllerStats,name=UncleanLeaderElectionsPerSec`
+* **Notification**:
+  * Notify when `UncleanLeaderElectionsPerSec > 0`
+  * Alarm when `UncleanLeaderElectionsPerSec > 0` for more than 1min.
+
+## Under Replicated Partitions
+* **Type**: Message Delivery
+* **Description**: In a healthy cluster, the number of in sync replicas (ISRs) should be exactly equal to the total number of replicas. In other words, the metric ensure the partitions are respecting the topic replication factor configuration. Partitions under replicated can appear when a cluster node is down.
+* **Metric**: JMX `kafka.server:type=ReplicaManager,name=UnderReplicatedPartitions`
+* **Notification**:
+  * Notify when `UnderReplicatedPartitions > 0`
+  * Alarm when `UnderReplicatedPartitions > 0` for more than 1min.
+
+## Under Min In Sync Replicas
+* **Type**: Message Delivery
+* **Description**:  If the cluster can't reach the `min.insync.replicas` and `acks` is set to all (because there is no enough broker up in the cluster for instance), the data producer can't receive the ack, it raises a timeout exception and may retry depending on its configuration). In a nutshell, when you have partition under min ISR is almost sure the the data production is blocked.
+* **Metric**: JMX `kafka.server:type=ReplicaManager,name=UnderMinIsrPartitionCount`
+* **Notification**: Alarm when `UnderMinIsrPartitionCount > 0`
+
+## Max lag by groupid by partition
+* **Type**: Performance
+* **Description**: The _lag_ is the delta between the offset of the last data appended in a topic and the offset of the committed read of a consumer. The bigger the lag is, the slower the consumer is reading the data.
+* **Metric**:
+* **Notification**: N/A
```

Comment:
Good starting point we should track here options to measure the consumer lag:

- on consumer side with JMX
- by using kafka-consumer-groups 
- by using external tools that provide such lag information